xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
I don't think so. I think we want to match the behaviour of bash and other shells. Text editors and other tools rely on the these command line options to work in a certain way and we want to have comparability with those tools. Changing `-i` from non-blocking to blocking would probably break a bunch of these things.